exporter/azuremonitorexporter/trace_to_envelope.go

```diff
@@ -0,0 +1,660 @@
+
+	return envelope, nil
+}
+
+// Maps Server/Consumer Span to AppInsights RequestData
+func spanToRequestData(span pdata.Span, incomingSpanType spanType) *contracts.RequestData {
+	// See https://github.com/microsoft/ApplicationInsights-Go/blob/master/appinsights/contracts/requestdata.go
+	// Start with some reasonable default for server spans.
+	data := contracts.NewRequestData()
+	data.Id = idToHex(span.SpanID())
+	data.Name = span.Name()
+	data.Duration = formatSpanDuration(span)
+	data.Properties = make(map[string]string)
+	data.Measurements = make(map[string]float64)
+	data.ResponseCode, data.Success = getDefaultFormattedSpanStatus(span.Status())
+
+	switch incomingSpanType {
+	case httpSpanType:
+		fillRequestDataHTTP(span, data)
+	case rpcSpanType:
+		fillRequestDataRPC(span, data)
+	case messagingSpanType:
+		fillRequestDataMessaging(span, data)
+	case faasSpanType:
+		// TODO Implement the FaaS mapping
+	}
+
+	return data
+}
+
+// Maps Span to AppInsights RemoteDependencyData
+func spanToRemoteDependencyData(span pdata.Span, incomingSpanType spanType) *contracts.RemoteDependencyData {
+	// https://github.com/microsoft/ApplicationInsights-Go/blob/master/appinsights/contracts/remotedependencydata.go
+	// Start with some reasonable default for dependent spans.
+	data := contracts.NewRemoteDependencyData()
+	data.Id = idToHex(span.SpanID())
+	data.Name = span.Name()
+	data.ResultCode, data.Success = getDefaultFormattedSpanStatus(span.Status())
+	data.Duration = formatSpanDuration(span)
+	data.Properties = make(map[string]string)
+	data.Measurements = make(map[string]float64)
+
+	switch incomingSpanType {
+	case httpSpanType:
+		fillRemoteDependencyDataHTTP(span, data)
+	case rpcSpanType:
+		fillRemoteDependencyDataRPC(span, data)
+	case databaseSpanType:
+		fillRemoteDependencyDataDatabase(span, data)
+	case messagingSpanType:
+		fillRemoteDependencyDataMessaging(span, data)
+	case faasSpanType:
+		// TODO Implement the FaaS mapping
+	}
+
+	return data
+}
+
+// Maps Span to AppInsights MessageData
+func spanToMessageData(span pdata.Span, incomingSpanType spanType) *contracts.MessageData {
+	// https://github.com/microsoft/ApplicationInsights-Home/blob/master/EndpointSpecs/Schemas/Bond/MessageData.bond
+	data := contracts.NewMessageData()
+	data.Properties = make(map[string]string)
+	data.Message = span.Name()
+
+	// TODO Is there a new version of the AppInsights contracts?
+	// The Measurements field is missing.
+	// For now, just copy all the string attributes into the message.properties.
+	if span.Attributes().Len() > 0 {
+		span.Attributes().ForEach(
+			func(k string, v pdata.AttributeValue) {
+				switch v.Type() {
+				case pdata.AttributeValueSTRING:
+					data.Properties[k] = v.StringVal()
+				}
+			})
+	}
+
+	return data
+}
+
+func getFormattedHTTPStatusValues(statusCode int64) (statusAsString string, success bool) {
+	if statusCode != 0 {
+		return strconv.FormatInt(statusCode, 10), statusCode <= 399
```

Comment:
`statuscode <= 299` is correct.

I also would check lower bound as well. statuscode >= 200 && statuscode <= 299

Reply:
right.